201-timeseriesinsights-environment-with-iothub/azuredeploy.json

```diff
@@ -0,0 +1,252 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "iotHubName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the source event hub."
+      }
+    },
+    "iotHubSku": {
+      "type": "string",
+      "defaultValue": "S1",
+      "metadata": {
+        "description": "The name of the IoT hub Sku."
+      }
+    },
+    "iotHubSkuTier": {
+      "type": "string",
+      "defaultValue": "Standard",
+      "metadata": {
+        "description": "The billing tier for the IoT hub."
+      }
+    },
+    "consumerGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the consumer group that the Time Series Insights service will use to read the data from the event hub. NOTE: To avoid resource contention, this consumer group must be dedicated to the Time Series Insights service and not shared with other readers."
+      }
+    },
+    "existingIoTHubResourceId": {
```

Comment:
follow this sample: https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/azuredeploy.json

for the new/existing pattern...  looks like you'll have the following changes:
- add a parameter for the users intent (new vs. existing)
- all the defaultValues should be values appropriate for creating a new resource (i.e. not empty strings)
- user should pass in the resourceGroup name and the resourceName instead of the resourceId and the template will construct the resourceID from those params